webapp/packages/core-notifications/src/NotificationsItem/NotificationsItem.tsx

```diff
@@ -7,24 +7,29 @@
  */
 import { observer } from 'mobx-react-lite';
 
-import { Snackbar } from '@cloudbeaver/core-blocks';
-import { useController } from '@cloudbeaver/core-di';
-import type { INotification } from '@cloudbeaver/core-events';
+import { Snackbar, useErrorDetails } from '@cloudbeaver/core-blocks';
+import { ENotificationType, type INotification } from '@cloudbeaver/core-events';
 
-import { NotificationItemController } from './NotificationItemController';
+import { FADE_TIMEOUT } from './FADE_TIMEOUT';
 
 interface Props {
   notification: INotification<any>;
 }
 
 export const NotificationsItem = observer<Props>(function Notification({ notification }) {
-  const controller = useController(NotificationItemController, notification);
+  const errorDetails = useErrorDetails(notification.details ?? null);
 
   if (notification.customComponent) {
     const Custom = notification.customComponent();
     return <Custom notification={notification} {...notification.extraProps} />;
   }
 
+  let closeDelay = 0;
+
+  if (notification.type !== ENotificationType.Error && !notification.persistent && notification.autoClose !== false) {
+    closeDelay = FADE_TIMEOUT;
+  }
+
```

Comment:
I suggest to add something like this to avoid component being overloaded with logic complicity

```
const calcDelay = (notification: Props['notification']) => {
  if (notification.type !== ENotificationType.Error && !notification.persistent && notification.autoClose !== false) {
    return FADE_TIMEOUT;
  }

  return 0;
};
```